src/components/Create/components/Allocation/Allocation.tsx

```diff
@@ -0,0 +1,84 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { FormItemInput } from 'models/formItemInput'
+import { Split } from 'models/splits'
+import { V2V3CurrencyOption } from 'models/v2v3/currencyOption'
+import { createContext, useContext } from 'react'
+import { AllocationItem } from './AllocationItem'
+import { AllocationList } from './AllocationList'
+import { useAllocation } from './hooks/Allocation'
+
+export type AllocationSplit = Split & { id: string }
+
+const DEFAULT_SET_CURRENCY_FN = () => {
+  console.error('AllocationContext.setCurrency called but no provider set')
+}
+
+const AllocationContext = createContext<{
+  allocations: AllocationSplit[]
+  totalAllocationAmount?: BigNumber
+  allocationCurrency?: V2V3CurrencyOption
+  addAllocation: (allocation: AllocationSplit) => void
+  removeAllocation: (id: string) => void
+  upsertAllocation: (allocation: AllocationSplit) => void
+  setCurrency: (currency: V2V3CurrencyOption) => void
+}>({
+  allocations: [],
+  addAllocation: () => {
+    console.error('AllocationContext.addAllocation called but no provider set')
+  },
+  removeAllocation: () => {
+    console.error(
```

Comment:
FACTS

Reply:
More useful than I thought it would be - saved me debugging many times